linux/Platform.c

```diff
@@ -557,6 +559,24 @@ void Platform_getPressureStall(const char* file, bool some, double* ten, double*
    fclose(fd);
 }
 
+void Platform_getFileDescriptors(double* used, double* max) {
+   *used = NAN;
+   *max = 65536;
+
+   FILE* fd = fopen(PROCDIR "/sys/fs/file-nr", "r");
+   if (!fd)
+      return;
+
+   unsigned long v1, v2, v3;
+   int total = fscanf(fd, "%lu %lu %lu", &v1, &v2, &v3);
+   if (total == 3) {
+      *used = v1;
+      *max = v3;
```

Comment:
I don't think this part is going to work very well, the difference between used and max is typically enormous - so much so that used is completely dwarfed... e.g. on my laptop:

```
$ cat /proc/sys/fs/file-nr
4192	0	9223372036854775807

```

We may want to use the max-seen-so-far as the maximum instead?

Reply:
We maybe want to detect if the value for max was set explicitly. If so, use the set value, otherwise I'm fine with detecting "enormous" values and show "max seen" instead.

Reply:
Maybe display the value in exponent notation if it's bigger than a million (or any other threshold)? (Or try to format it as `2**63-1`?)

Reply:
I plan on limiting the maximum number displayed to about 2^30, as that many open files can effectively be seen as unlimited due to the amount of memory required by the kernel. Patch for this not implemented yet, due to lack of time …

Reply:
@BenBE thanks, like this aspect is resolved in the non-platform code now - looks good.